packages/twitch-chat-client/src/Toolkit/UserTools.ts

```diff
@@ -1,7 +1,15 @@
 /** @private */
 export function toUserName(channel: string): string {
 	// it's okay if this is already a user name, we only remove the first character if it's a pound
-	return channel.replace(/^#/, '').toLowerCase();
+	const name = channel.replace(/^#/, '').toLowerCase();
+	// valid usernames must be alphanumberic or underscores between 4 and 25 characters (cannot start with underscore)
+	const re = RegExp(/^[a-zA-Z0-9][a-zA-Z0-9_]{3,24}$/);
+	if (!re.test(name)) {
+		throw new Error(
+			'Invalid channel or username. It must be made from alphanumberic characters and between 4 and 25 characters in length. Usernames and channels names cannot start with an underscore.'
```

Comment:
It would make sense to output the offending user name.

Something like:

```js
`"$[name}" is not a valid user or channel name. It must be 4-25 characters long, can only include letters, numbers and underscores, and can not start with an underscore.`
```